app/controllers/saved_scenario_reports_controller.rb

```diff
@@ -0,0 +1,52 @@
+# Create a report based on a .yml file in config/saved_scenario_reports
+# The report name and format should be provided in params (only works for csv)
+class SavedScenarioReportsController < ApplicationController
+  before_action :ensure_valid_browser
+
+  def show
+    @scenario_id = SavedScenario.find(params[:saved_scenario_id]).scenario_id
+    @yml = YAML.load_file("config/saved_scenario_reports/#{report_name}.yml")
+
+    if valid_api_response? && valid_report_name?
+      @query_api_response = api_response['gqueries']
+      respond_to do |format|
+        format.csv { render 'saved_scenarios/reports/show.csv.erb' }
```

Comment:
The Rails convention is to put views in a directory which matches the controller name. So rather than placing the show.csv.erb file in app/views/saved_scenarios/reports, it would instead go in app/views/saved_scenario_reports. As a bonus, the `respond_to` block can then be simpler as Rails will find the file automatically.

```suggestion
        format.csv
```